tests/unit/test_bugjira.py

```diff
@@ -0,0 +1,180 @@
+    assert broker == sandboxed_bugjira._bugzilla_broker
+
+
+def test_get_broker_with_jira_issue_id(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance's _get_broker method
+    WHEN we call it with a key string that represents a jira issue key
+    THEN the instance's _jira_broker attribute should be returned
+    """
+    broker = sandboxed_bugjira._get_broker("FOO-123")
+    assert broker == sandboxed_bugjira._jira_broker
+
+
+def test_get_issue_with_non_string_key(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance's get_issue method
+    WHEN we call it with a non-string parameter
+    THEN a ValueError is raised
+    """
+    with pytest.raises(ValueError):
+        sandboxed_bugjira.get_issue(1)
+
+
+def test_get_issue_good_bugzilla(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we invoke the get_issue method with a BZ number
+    THEN it should return an instance of Issue with a non-None bugzilla
+        attribute and a key matching the input
+    AND the instance's bugzilla attribute should have had its getbug method
+        invoked once
+    """
+    key = "123456"
+    issue = sandboxed_bugjira.get_issue(key)
+    assert isinstance(issue, BugzillaIssue)
+    assert issue.bugzilla
+    assert not issue.jira_issue
+    assert issue.key == key
+    assert sandboxed_bugjira.bugzilla.getbug.call_count == 1
+
+
+def test_get_issue_bad_bugzilla(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we invoke the get_issue method with a BZ number using a bug id that is
+        known not to correspond to any bug on the remote bugzilla service
+    THEN a Fault exception should be raised
+    """
+    sandboxed_bugjira.bugzilla.getbug.side_effect = Fault(
+        "Fault 101", "Bug #XXX does not exist."
+    )
+    with pytest.raises(Fault):
+        sandboxed_bugjira.get_issue("123")
+
+
+def test_get_issue_good_jira_issue(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we invoke the get_issue method with a Jira issue key
+    THEN it should return an instance of Issue with a non-None jira_issue
+        attribute and a key matching the input
+    AND the instance's jira attribute should have had its issue method invoked
+        once
+    """
+    key = "FOO-123"
+    issue = sandboxed_bugjira.get_issue(key)
+    assert isinstance(issue, JiraIssue)
+    assert issue.jira_issue
+    assert not issue.bugzilla
+    assert issue.key == key
+    assert sandboxed_bugjira.jira.issue.call_count == 1
+
+
+def test_get_issue_bad_jira_issue(sandboxed_bugjira):
+    """
+    GIVEN a Bugjira instance
+    WHEN we invoke the get_issue method with a JIRA issue key using a key that
+        is known not to correspond to any issue on the remote bugzilla service
+    THEN a JIRAError exception should be raised
+    """
+    sandboxed_bugjira.jira.issue.side_effect = JIRAError
+    with pytest.raises(JIRAError):
+        sandboxed_bugjira.get_issue("FOO-666")
```

Comment:
Another example we might consider wrapping. If we had something like BugJiraIssueException that this method might throw, a caller could always check for that and inspect the details, which we would surface that way.

Reply:
Agreed. We should have our own exceptions, even if it's just a generic one for now.